cabot/metricsapp/api/grafana_elastic.py

```diff
@@ -320,6 +321,26 @@ def adjust_time_range(queries, time_range):
     minimum = '{}m'.format(time_range)
 
     for n, query in enumerate(queries):
+        # Minimum adjustment for derived metrics
+        date_histogram = _get_date_histogram(query)
+
+        aggs = query['aggs']
+        # Try to find derivative or moving average to extend bounds if necessary. This avoids situations where
+        # we use partial/empty buckets to calculate derivates/moving averages.
+        while aggs.get('agg'):
+            aggs = aggs['agg']['aggs']
```

Comment:
Is derivative at top level of `query['aggs']` a valid query/something we need to support? Something like:

```json
{
  "aggs": {
    "derivative": "blah"
  }
}
```
(in this case the while would never trigger; this probably isn't a real case, but thought I'd bring it up)

Reply:
The first level aggregation is always `date_histogram` so there's no chance of a metric being under `query['aggs']`